File: TheLionsDen.Model/Responses/ReservationResponse.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace TheLionsDen.Model.Responses
{
    public class ReservationResponse
    {
        public int ReservationId { get; set; }
        public DateTime Arrival { get; set; }
        public DateTime Departure { get; set; }
        public float TotalPrice { get; set; }
        public string Status { get; set; }
        public string SpecialRequests { get; set; }
        public string EstimatedArrivalTime { get; set; }
        public int UserId { get; set; }
        public int RoomId { get; set; }
        public int PaymentDetailsId { get; set; }

        public string RoomName { get; set; }
        public string UserFullName { get; set; }
        public string FacilityNames { get; set; }

        public virtual PaymentDetailResponse PaymentDetails { get; set; }
        public virtual RoomResponse Room { get; set; }
        public virtual UserResponse User { get; set; }
        [JsonIgnore]
        public virtual ICollection<ReservationFaciliteResponse> ReservationFacilites { get; set; }
    }
}

File: the_lions_den_mobile/lib/model/reservation/reservation_response.dart
import 'package:json_annotation/json_annotation.dart';
part 'reservation_response.g.dart';

@JsonSerializable()
class ReservationResponse {
  int? reservationId;
  double? totalPrice;
  DateTime? arrival;
  DateTime? departure;
  String? specialRequests;
  String? estimatedArrivalTime;
  String? status;
  int? userId;
  int? roomId;

  String? roomName;
  String? userFullName;
  String? facilityNames;

  ReservationResponse();

  factory ReservationResponse.fromJson(Map<String, dynamic> json) =>
      _$ReservationResponseFromJson(json);

  /// Connect the generated [_$ReservationResponseToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$ReservationResponseToJson(this);
}
